easybuild/easyblocks/p/psmpi.py

```diff
@@ -92,8 +93,14 @@ def configure_step(self):
             self.cfg.update('configopts', ' --with-mpichconf="%s"' % self.cfg['mpich_opts'])
 
         # Lastly, set pscom related variables
-        self.cfg.update('preconfigopts', ('PSCOM_LDFLAGS=-L{0}/lib '+
-                'PSCOM_CPPFLAGS=-I{0}/include').format(get_software_root('pscom')))
+        if self.cfg['pscom_allin_path'] is None:
+            self.cfg.update('preconfigopts', ('PSCOM_LDFLAGS=-L{0}/lib '+
+                    'PSCOM_CPPFLAGS=-I{0}/include').format(get_software_root('pscom')))
+        else:
+            pscom_allin_path = self.cfg['pscom_allin_path'].strip()
+            self.cfg.update('configopts', ' --with-pscom-allin="%s"' % pscom_allin_path)
+            self.cfg.update('preconfigopts', ('PSCOM_LDFLAGS=-L{0}/lib '+
+                    'PSCOM_CPPFLAGS=-I{0}/include').format(pscom_allin_path))
```

Comment:
I would reformat this a bit for readability, and to avoid duplicate code:

```python
if self.cfg['pscom_allin_path'] is None:
    pscom_allin_path = get_software_root('pscom')
else:
    pscom_allin_path = self.cfg['pscom_allin_path'].strip()
    self.cfg.update('configopts', ' --with-pscom-allin="%s"' % pscom_allin_path)

pscom_flags = "PSCOM_LDFLAGS=-L{0}/lib PSCOM_CPPFLAGS=-I{0}/include".format(pscom_allin_path)
self.cfg.update('preconfigopts', pscom_flags)
```

Reply:
Done in the next commit